File: medium/cpp/1524.cpp
// https://leetcode.com/problems/number-of-sub-arrays-with-odd-sum/

class Solution
{
private:
    static constexpr auto mod = 1000'000'000 + 7;

public:
    int numOfSubarrays(std::vector<int>& arr)
    {
        auto odd = 0;
        auto even = 1;
        auto res = 0;
        auto s = 0LL;

        for (auto e : arr)
        {
            s += e;

            if (s % 2)
            {
                res += even;
                odd++;
            }
            else
            {
                res += odd;
                even++;
            }

            res %= mod;
        }

        return res;
    }
};

File: medium/python/1524.py
# https://leetcode.com/problems/number-of-sub-arrays-with-odd-sum/

class Solution:
    def numOfSubarrays(self, arr: List[int]) -> int:
        odd, even = 0, 1
        res = 0
        sum = 0

        for e in arr:
            sum += e

            if sum % 2 == 1:
                res += even
                odd += 1
            else:
                res += odd
                even += 1

            res %= 1000000007

        return res
        